src/Raven.Server/RavenServer.cs

```diff
@@ -931,39 +937,105 @@ internal AuthenticateConnection AuthenticateConnectionCertificate(X509Certificat
                 using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
                 using (ctx.OpenReadTransaction())
                 {
-                    var certKey = Constants.Certificates.Prefix + certificate.Thumbprint;
-                    var cert = ServerStore.Cluster.Read(ctx, certKey) ??
-                               ServerStore.Cluster.GetLocalState(ctx, certKey);
-
-                    if (cert == null)
-                    {
-                        authenticationStatus.Status = AuthenticationStatus.UnfamiliarCertificate;
-                    }
-                    else
+                    try
                     {
-                        var definition = JsonDeserializationServer.CertificateDefinition(cert);
-                        authenticationStatus.Definition = definition;
-                        if (definition.SecurityClearance == SecurityClearance.ClusterAdmin)
-                        {
-                            authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
-                        }
-                        else if (definition.SecurityClearance == SecurityClearance.ClusterNode)
-                        {
-                            authenticationStatus.Status = AuthenticationStatus.ClusterAdmin;
-                        }
-                        else if (definition.SecurityClearance == SecurityClearance.Operator)
+                        var certKey = Constants.Certificates.Prefix + certificate.Thumbprint;
+                        var cert = ServerStore.Cluster.Read(ctx, certKey) ??
+                                   ServerStore.Cluster.GetLocalState(ctx, certKey);
+
+                        if (cert == null)
                         {
-                            authenticationStatus.Status = AuthenticationStatus.Operator;
+                            // The certificate is not explicitly registered in our server, let's see if we have a certificate
+                            // with the same public key pinning hash.
+                            var allCertKeys = ServerStore.Cluster.ItemKeysStartingWith(ctx, Constants.Certificates.Prefix, 0, int.MaxValue).ToList();
```

Comment:
No. We may have a lot of them, and we don't want to process them by hand.
Instead, add a Voron index for the public key pinning hash and check on that.

Reply:
Certificate definitions are stored in the "Items" table along with other objects. Do you want to define an index on that table or move the certs to a table of their own?

Reply:
Yes, given that we have new behavior for that, we should store it separately and have a dedicated index for that.